drivers/gpio/gpio_creg_gpio.c

```diff
@@ -119,6 +112,50 @@ static int pin_interrupt_configure(const struct device *dev,
 	return -ENOTSUP;
 }
 
+static int pin_config(const struct device *dev,
```

Comment:
I'd split up this patch in 2, first being improving the creg_gpio driver (so far, pin_config() implementation is the driver improvement. The spi cs is just the use case that triggered the requirement for such improvement).

And second patch for the em_starterkit max freq. (In the commit message, explain why you reduced such freq btw)

Reply:
Agreed.

Reply:
Thanks for the suggestion, now updated.